src/inmanta/plugins.py

```diff
@@ -277,26 +281,32 @@ def to_dsl_type(python_type: type[object]) -> inmanta_type.Type:
 
         # dict
         if issubclass(origin, collections.abc.Mapping):
-            args = typing_inspect.get_args(python_type)
-            if not args:
-                return inmanta_type.TypedDict(inmanta_type.Type())
+            if origin in [collections.abc.Mapping, dict, typing.Mapping]:
+                args = typing_inspect.get_args(python_type)
+                if not args:
+                    return inmanta_type.TypedDict(inmanta_type.Type())
 
-            if not issubclass(args[0], str):
-                raise TypingException(
-                    None, f"invalid type {python_type}, the keys of any dict should be 'str', got {args[0]} instead"
-                )
+                if not issubclass(args[0], str):
+                    raise TypingException(
+                        None, f"invalid type {python_type}, the keys of any dict should be 'str', got {args[0]} instead"
+                    )
 
-            if len(args) == 1:
-                return inmanta_type.TypedDict(inmanta_type.Type())
+                if len(args) == 1:
+                    return inmanta_type.TypedDict(inmanta_type.Type())
 
-            return inmanta_type.TypedDict(to_dsl_type(args[1]))
+                return inmanta_type.TypedDict(to_dsl_type(args[1]))
+            else:
+                raise TypingException(None, f"invalid type {python_type}, dictionary types should be Mapping or dict")
```

Comment:
Nice. I hadn't considered this, but it's good to have a clear message!

One suggestion
```suggestion
                raise TypingException(None, f"invalid type {python_type}, dictionary types should be Mapping (preferred) or dict")
```